src/bot.js

```diff
@@ -165,6 +165,46 @@ function addLFG(MESSAGE) {
         });
 }
 
+function endSession(message){
+    var author = message.author,
+        guild_id = message.guild.id,
+        role_id = '',
+        roles = message.member.roles.array(),
+        role = null;
+
+    for(var i = 0; i < roles.length; i++){
+        role = roles[i];
```

Comment:
Oh, you're right on that. The issue actually is that it continues the loop and sets `role` to the next one(s), so if it finds an lfg role, it deletes a different role.